### PR TITLE
Separate typings for node and browser

### PR DESCRIPTION
**Pull request by frenzzy**
## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

* [x] My code follows the code style of this project.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [x] I have read the **CONTRIBUTING** document.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,39 +17,40 @@
   ],
   "main": "node/index.js",
   "module": "node/module.js",
-  "typings": "render.d.ts",
+  "typings": "src/node.d.ts",
```
Comment (joseluisq) [inline package.json]: Why `"typings": "src/node.d.ts"`? Why not `"typings": "src/index.d.ts"`

Comment (joseluisq) [reply package.json]: Well, ignore it. `src/node.d.ts` makes more sense.

Comment (frenzzy) [reply package.json]: - `index.{js,d.ts}` - common code for node and browser which exports internal `renderer` and `renderToString`
- `browser.{js,d.ts}` - client-side code which re-exports `renderToString` from index and exports `withRender` with `toString` only
- `node.{js,d.ts}` - server-side code which re-exports `renderToString` from index plus `renderToStream` and also exports `withRender` with `toString` and `toStream`

Comment (joseluisq) [reply package.json]: In short, `App` is the type for one `app()` function that it will need  `withRender()` as param.

Comment (frenzzy) [reply package.json]: Sounds reasonable, will you create a PR to Hyperapp repo or should I?

Comment (joseluisq) [reply package.json]: Sure, I will send a PR explaining about that.

Comment (joseluisq) [reply package.json]: > index.{js,d.ts} - common code for node and browser which exports internal renderer and renderToString
> browser.{js,d.ts} - client-side code which re-exports renderToString from index and exports withRender with toString only
> node.{js,d.ts} - server-side code which re-exports renderToString from index plus renderToStream and also exports withRender with toString and toStream

Great!

Comment (joseluisq) [reply package.json]: Only a detail about importing.
Here some examples:

```ts
// 1. Testing:
import { renderToStream, withRender, Render, renderToString } from "../../src/node"
// 2. Actually: if I want to import node version
import { renderToStream, withRender, Render, renderToString } from "@hyperapp/render/src/node"
// 3. Purpose: move typings to root directory
import { renderToStream, withRender, Render, renderToString } from "@hyperapp/render/node"
```

Why do you think about the third option?

Comment (frenzzy) [reply package.json]: Both 2 and 3 options are valid after this PR.

Comment (joseluisq) [reply package.json]: fine!

Comment (joseluisq) [reply package.json]: The PR was sent..

**Pull request merged**